### DISCUSSION TITLE
Fetch all records with or without many-many relationship

### DISCUSSION THREAD

**Discussion by mbahArip** [supabase/supabase#18888] [Questions]
Hello all,
So I'm trying to create a posts table with tags, but I'm having trouble with fetching the posts without the tags.
Currently I'm using `!inner` on the tags since I want to filter the posts based on the tags selected.

My post flow are like this.
`User click new post > Insert a new post with status as draft > User edit the content > User either publish the post or keep the post as draft`

But it won't fetch post data without the tag relations.
Is there a way to be able to fetch the post without tags and keep the filter tags function?

Table structure and my current code below:

### posts
```
id: uuid;
created_at: timestampz;
title: text;
summary?: text;
content?: text;
slug?: text;
status: status; // my custom enum type
category_id: uuid;
```

### master_tag
```
id: uuid;
created_at: timestampz;
title: text;
description?: text;
slug?: text;
color: text;
```

### rel_post_tag
```
post_id: uuid;
tag_id: uuid;
```

## Current Code
```ts
const buildFetch = supabase
  .from( 'posts' )
  .select( '*, category:master_category(*), tags:master_tag!inner(*)', {
    count: 'exact'
  } );

if (query.tags) buildFetch.in( 'tags.id', query.tags.split(',') )
```

**Comment by GaryAustin1** [comment]
If you don't want the parent table filtered why are you using !inner.
https://supabase.com/docs/reference/javascript/select?example=filtering-through-foreign-tables
Also you can do multiple operation on the same join table by using aliases.

**Reply by mbahArip** [reply]
Sorry if my explanation is confusing. (I hope I explain it better this time)

What I mean is, for example if I have an endpoint on `/api/admin/posts`
I want to fetch all posts data wether they have an tag relation or not.

But if i want to filter it by tags (ex: tags1 & tags2), the endpoint will be `/api/admin/posts?tags=tags1,tags2`
With filter on, I want to exclude all posts that doesn't have this tags.

I've already tried the example on docs, but it still returning the post data that doesn't have the tags. (I've tried both eq and in filter, it still the same).
and the reason I'm using !inner hint since I want it to be filtered out if there are tags query.
